components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,133 @@
+import { Dropdown } from "components";
+import { grey800, white } from "constants/colors";
+import { addOpacityToHsl } from "helpers";
+import { usePaginationContext } from "hooks";
+import PreviousPage from "public/assets/icons/left-chevron.svg";
+import NextPage from "public/assets/icons/right-chevron.svg";
+import styled, { CSSProperties } from "styled-components";
+import { DropdownItemT, PaginateForT } from "types";
+
+export interface Props {
+  paginateFor: PaginateForT;
```

Comment:
Pagination logic all lives in the pagination context, keyed by name (see type definition for `PaginateForT`.

This means that any component can access pagination information, while still keeping the states independent from one another.